Antidote/CallViewController.m

```diff
@@ -0,0 +1,184 @@
+//
+//  CallViewController.m
+//  Antidote
+//
+//  Created by Chuong Vu on 7/1/15.
+//  Copyright (c) 2015 dvor. All rights reserved.
+//
+
+#import "CallViewController.h"
+#import "Masonry.h"
+#import "Helper.h"
+#import "OCTCall.h"
+#import "OCTChat.h"
+#import "OCTSubmanagerCalls.h"
+
+NSString *const kCallViewControllerUserIdentifier = @"callViewController";
+
+@interface CallViewController () <RBQFetchedResultsControllerDelegate>
+
+@property (strong, nonatomic) OCTSubmanagerCalls *manager;
+@property (strong, nonatomic, readwrite) OCTChat *chat;
+@property (strong, nonatomic) OCTCall *call;
+@property (strong, nonatomic) RBQFetchedResultsController *callController;
+
+@property (strong, nonatomic) UILabel *nameLabel;
+@property (strong, nonatomic) UIButton *cancelCallButton;
+@property (strong, nonatomic) UILabel *timerLabel;
+
+@end
+
+@implementation CallViewController
+
+#pragma mark - Life cycle
+
+- (instancetype)initWithChat:(OCTChat *)chat submanagerCalls:(OCTSubmanagerCalls *)manager
+{
+    self = [super init];
+    if (! self) {
+        return nil;
+    }
+
+    _chat = chat;
+
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"chat == %@", chat];
+    _callController = [Helper createFetchedResultsControllerForType:OCTFetchRequestTypeCall predicate:predicate delegate:self];
+    _callController.delegate = self;
+
+    _call = [_callController fetchedObjects].firstObject;
+    if (! _call) {
+        return nil;
+    }
+
+    _manager = manager;
+
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+
+    self.view.opaque = YES;
+
+    [self setupBlurredView];
+    [self createNameLabel];
+
+    [self createEndCallButton];
+    [self createCallTimer];
+
+    [self layoutSubviews];
```

Comment:
> You should not call this method directly. If you want to force a layout update, call the setNeedsLayout method instead to do so prior to the next drawing update. If you want to update the layout of your views immediately, call the layoutIfNeeded method.

[link](https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIView_Class/#//apple_ref/occ/instm/UIView/layoutSubviews)

Also there is no need in asking for layout, it will happen automatically after viewDidLoad.